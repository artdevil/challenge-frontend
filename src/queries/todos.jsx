import { gql } from 'react-apollo';

const queries = {
  TodosListQuery: gql`
    query TodosListQuery {
      todos {
        id
        title
        completed
      }
    }
  `,
};

export default queries;
