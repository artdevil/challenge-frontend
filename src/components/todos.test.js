import React from 'react';
import { shallow } from 'enzyme';
import { ApolloClient, ApolloProvider } from 'react-apollo';
import { mockNetworkInterface } from 'react-apollo/test-utils';
import queries from '../queries/todos';

import Todos from './todos';

it('shallow todos renders without crashing', () => {
  const data = { todos: { id: 1, label: 'hello world', completed: true } };

  const networkInterface = mockNetworkInterface(
    { request: { query: queries.TodosListQuery }, result: { data } },
  );

  const client = new ApolloClient({ networkInterface, addTypename: false });

  const wrapper = shallow(
    <ApolloProvider client={client}>
      <Todos />
    </ApolloProvider>);

  expect(wrapper).toMatchSnapshot();
});
