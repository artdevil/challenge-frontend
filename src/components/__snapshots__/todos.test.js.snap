// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow todos renders without crashing 1`] = `
<QueryRecyclerProvider>
  <Apollo(Apollo(Apollo(Todos))) />
</QueryRecyclerProvider>
`;
